src/features/packages/pack/zipUtils.ts

```diff
@@ -53,8 +53,12 @@ export async function doZip(
     setCompressionLevel(zip, compressionLevel || 8);
 
     process.chdir(initialWorkingDirectory);
+    if (fs.existsSync(archivePath) && overwrite === false)

```

Comment:
The old version `{ overwrite: overwrite || true }` would always resolve to true, this doesn't seem like the intended behaviour. I'm a little torn on changing this as it is potentially a change of existing behaviour but this feels like the correct solution long term.

Reply:
That change was introduced in [this seemingly unrelated PR](https://github.com/OctopusDeploy/api-client.ts/commit/155d62964ba4a2aeba4cd7fa118738cba53af1df) with no comments around it

Reply:
I think you are correct, setting overwrite to `false` would resolve to `true` and overwrite anyway.
You could raise an issue and link/fix it to this PR for the behaviour change